config/webpack.config.js

```diff
@@ -0,0 +1,212 @@
+};
+
+// ------------------------------------
+// Bundle Output
+// ------------------------------------
+webpackConfig.output = {
+  filename   : `[name].[${project.compiler_hash_type}].js`,
+  path       : project.paths.dist(),
+  publicPath : project.compiler_public_path
+};
+
+// ------------------------------------
+// Plugins
+// ------------------------------------
+webpackConfig.plugins = [
+  // DefinePlugin is used to define global variables that can be used in the app.
+  new webpack.DefinePlugin(project.globals),
+  // HtmlWebpackPlugin injects the bundled assets into the entry HTML file.
+  new HtmlWebpackPlugin({
+    template : project.paths.client('index.html'),
+    hash     : false,
+    favicon  : project.paths.public('favicon.ico'),
+    filename : 'index.html',
+    inject   : 'body',
+    minify   : {
+      collapseWhitespace : true
+    }
+  })
+];
+
+// Ensure that the compiler exits on errors during testing so that
+// they do not get skipped and misreported.
+if (__TEST__ && !argv.watch) {
+  webpackConfig.plugins.push(function () {
+    this.plugin('done', function (stats) {
+      if (stats.compilation.errors.length) {
+        // Pretend no assets were generated. This prevents the tests
+        // from running making it clear that there were warnings.
+        throw new Error(
+          stats.compilation.errors.map(err => err.message || err)
+        )
+      }
+    })
+  })
+}
+
+if (__DEV__) {
+  debug('Enabling plugins for live development (HMR, NoErrors).');
+  webpackConfig.plugins.push(
+    new webpack.HotModuleReplacementPlugin(),
+    new webpack.NoErrorsPlugin()
+  )
+} else if (__PROD__) {
+  debug('Enabling plugins for production (OccurenceOrder, Dedupe & UglifyJS).');
+  webpackConfig.plugins.push(
+    new webpack.optimize.OccurrenceOrderPlugin(),
+    new webpack.optimize.DedupePlugin(),
+    new webpack.optimize.UglifyJsPlugin({
+      compress : {
+        unused    : true,
+        dead_code : true,
+        warnings  : false
+      }
+    })
+  )
+}
+
+// Don't split bundles during testing, since we only want import one bundle
+if (!__TEST__) {
+  webpackConfig.plugins.push(
+    new webpack.optimize.CommonsChunkPlugin({
+      names : ['vendor']
+    })
+  )
+}
+
+// ------------------------------------
+// Loaders
+// ------------------------------------
+// JavaScript / JSON
+webpackConfig.module.loaders = [{
+  test    : /\.(js|jsx)$/,
+  exclude : /node_modules/,
+  loader  : 'babel',
+  query   : project.compiler_babel
+}, {
+  test   : /\.json$/,
+  loader : 'json'
+}];
+
+// ------------------------------------
+// Style Loaders
+// ------------------------------------
+// We use cssnano with the postcss loader, so we tell
+// css-loader not to duplicate minimization.
+const BASE_CSS_LOADER = 'css?sourceMap&-minimize';
+
+webpackConfig.module.loaders.push({
+  test    : /\.scss$/,
+  exclude : null,
+  loaders : [
+    'style',
+    BASE_CSS_LOADER,
+    'postcss',
```

Comment:
Currently the main thing it's used for is autoprefixing css for cross-browser support (https://github.com/postcss/autoprefixer).

It's also handling the minification of the CSS here - I think it does a better job of it than the css-loader does.

And to be clear - this isn't doing any CSS Modules stuff - I think it's most definitely worth it to have the autoprefixer here, and it also leaves us open for including additional postcss addons in the future